robot/impl/resource_manager.go

```diff
@@ -194,15 +216,14 @@ func (manager *resourceManager) updateRemoteResourceNames(
 
 	for _, resName := range newResources {
 		remoteResName := resName
+		resLogger := logger.WithFields("resource", remoteResName)
```

Comment:
flyby: similar to the [flyby above](https://github.com/viamrobotics/rdk/pull/4421/files#r1788260478) - all logs local to this loop should include in the resource name in the current loop variable.